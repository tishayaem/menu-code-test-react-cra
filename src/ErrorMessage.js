import React from 'react';

export default function ErrorMessage(props) {
  
    return <div className="message">{props.error}</div>;
  }
