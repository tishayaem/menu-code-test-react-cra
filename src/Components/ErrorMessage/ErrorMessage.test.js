import React from 'react';
import ErrorMessage from './ErrorMessage';

describe('ErrorMessage', () => {
  const message =
    "We are sorry, but you can't have Salmon fillet with prawn cocktail!";
  const wrapper = shallow(<ErrorMessage error={message} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the correct message', () => {
    expect(wrapper.find('.message').text()).toMatch(message);
  });

});
