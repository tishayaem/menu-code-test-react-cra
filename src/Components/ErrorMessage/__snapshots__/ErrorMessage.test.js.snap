// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage
    error="We are sorry, but you can't have Salmon fillet with prawn cocktail!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "We are sorry, but you can't have Salmon fillet with prawn cocktail!",
      "className": "message",
    },
    "ref": null,
    "rendered": "We are sorry, but you can't have Salmon fillet with prawn cocktail!",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "We are sorry, but you can't have Salmon fillet with prawn cocktail!",
        "className": "message",
      },
      "ref": null,
      "rendered": "We are sorry, but you can't have Salmon fillet with prawn cocktail!",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
